pkg/migrations/sql/idempotent/base.sql

```diff
@@ -2154,6 +2154,146 @@ SET search_path = pg_temp;
+        SELECT
+            x.node_name,
+            _timescaledb_internal.ping_data_node(x.node_name) AS node_up
+        FROM (
+            SELECT DISTINCT dht.node_name -- only ping each node once
+            FROM dht
+        ) x
+    )
+    SELECT
+        dht.table_name as hypertable_name,
+        dht.node_name,
+        up.node_up
+    FROM dht
+    JOIN up ON (dht.node_name = up.node_name)
+$function$
+LANGUAGE sql
+STRICT STABLE 
+SECURITY DEFINER
+--search path must be set for security definer
+SET search_path = pg_temp;
+REVOKE ALL ON FUNCTION SCHEMA_CATALOG.hypertable_node_up(name) FROM PUBLIC;
+GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.hypertable_node_up(name) to prom_modifier;
+
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.hypertable_remote_size(schema_name_in name)
+RETURNS TABLE(hypertable_name name, node_name name, table_bytes bigint, index_bytes bigint, toast_bytes bigint, total_bytes bigint)
+AS $function$
+    SELECT
+        dht.hypertable_name,
+        dht.node_name,
+        sum(x.table_bytes)::bigint AS table_bytes,
+        sum(x.index_bytes)::bigint AS index_bytes,
+        sum(x.toast_bytes)::bigint AS toast_bytes,
+        sum(x.total_bytes)::bigint AS total_bytes
+    FROM SCHEMA_CATALOG.hypertable_node_up(schema_name_in) dht
+    LEFT OUTER JOIN LATERAL _timescaledb_internal.data_node_hypertable_info(
+        CASE WHEN dht.node_up THEN
+            dht.node_name
+        ELSE
+            NULL
+        END, schema_name_in, dht.hypertable_name) x ON true
+    GROUP BY dht.hypertable_name, dht.node_name
+$function$
+LANGUAGE sql
+STRICT STABLE 
+SECURITY DEFINER
+--search path must be set for security definer
+SET search_path = pg_temp;
+REVOKE ALL ON FUNCTION SCHEMA_CATALOG.hypertable_remote_size(name) FROM PUBLIC;
+GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.hypertable_remote_size(name) to prom_modifier;
+
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.hypertable_compression_stats(schema_name_in name)
+RETURNS TABLE(hypertable_name name, total_chunks bigint, number_compressed_chunks bigint, before_compression_total_bytes bigint, after_compression_total_bytes bigint)
+AS $function$
+    SELECT
+        ch.hypertable_name,
+        count(*)::bigint AS total_chunks,
+        (count(*) FILTER (WHERE ch.compression_status = 'Compressed'))::bigint AS number_compressed_chunks,
+        sum(ch.before_compression_total_bytes)::bigint AS before_compression_total_bytes,
+        sum(ch.after_compression_total_bytes)::bigint AS after_compression_total_bytes
+    FROM
+    (
+        -- local hypertables
+        SELECT
+            ch.hypertable_name,
+            ch.compression_status,
+            ch.uncompressed_total_size AS before_compression_total_bytes,
+            ch.compressed_total_size AS after_compression_total_bytes,
+            NULL::text AS node_name
+        FROM _timescaledb_internal.compressed_chunk_stats ch
+        WHERE ch.hypertable_schema = schema_name_in
+        UNION ALL
```

Comment:
we use a UNION ALL for local and distributed for compression_stats but not sizes. is there a reason to treat these differently?

Reply:
The code in hypertable_local_size takes 13 seconds by itself in the test db when taking advantage of parallelism. Whenever I combined it with the hypertable_remote_size using UNION ALL, the planner stopped using parallel workers, and the runtime shot up to over an hour.
I can try it one more time now that they are in functions. Maybe that makes a difference.

Reply:
Retried a few different options. If I keep the hypertable_local_size logic and hypertable_remote_size logic in separate functions, parallelism is used and it runs quickly. If I put them together in the same function, the planner won't use parallelism.